include/alpaka/mem/buf/cpu/Fill.hpp

```diff
@@ -0,0 +1,146 @@
+
+            template<typename TViewFwd>
+            TaskFillCpuBase(TViewFwd&& view, TValue const& value, TExtent const& extent)
+                : m_value(value)
+                , m_extent(getExtents(extent))
+                , m_extentWidth(getExtents(extent).back())
+#if(!defined(NDEBUG)) || (ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL)
+                , m_dstExtent(getExtents(view))
+#endif
+                , m_dstPitchBytes(getPitchesInBytes(view))
+                , m_dstMemNative(reinterpret_cast<std::uint8_t*>(getPtrNative(view)))
+            {
+                ALPAKA_ASSERT((castVec<DstSize>(m_extent) <= m_dstExtent).all());
+                if constexpr(TDim::value > 1)
+                    ALPAKA_ASSERT(
+                        m_extentWidth * static_cast<ExtentSize>(sizeof(Elem)) <= m_dstPitchBytes[TDim::value - 2]);
+
+                ALPAKA_ASSERT(reinterpret_cast<std::uintptr_t>(m_dstMemNative) % alignof(Elem) == 0);
+            }
+
+            TValue const m_value;
+            Vec<TDim, ExtentSize> const m_extent;
+            ExtentSize const m_extentWidth;
+#if(!defined(NDEBUG)) || (ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL)
+            Vec<TDim, DstSize> const m_dstExtent;
+#endif
+            Vec<TDim, DstSize> const m_dstPitchBytes;
+            std::uint8_t* const m_dstMemNative;
+        };
+
+        // Generic ND version
+        template<typename TDim, typename TView, typename TExtent, typename TValue>
+        struct TaskFillCpu : public TaskFillCpuBase<TDim, TView, TExtent, TValue>
+        {
+            using Base = TaskFillCpuBase<TDim, TView, TExtent, TValue>;
+            using Base::Base;
+            using Elem = typename Base::Elem;
+            using ExtentSize = typename Base::ExtentSize;
+
+            // using DstSize = typename Base::DstSize;
+            // using DimMin1 = DimInt<TDim::value - 1u>;
+
+            ALPAKA_FN_HOST auto operator()() const -> void
+            {
+                // Vec<DimMin1, ExtentSize> const extentWithoutInnermost = subVecBegin<DimMin1>(this->m_extent);
+                // Vec<DimMin1, DstSize> const pitchBytesWithoutOutmost = subVecBegin<DimMin1>(this->m_dstPitchBytes);
+
+                if(static_cast<std::size_t>(this->m_extent.prod()) != 0u)
+                {
+                    meta::ndLoopIncIdx(
+                        this->m_extent,
+                        [&](Vec<TDim, ExtentSize> const& idx)
+                        {
+                            std::uintptr_t offsetBytes
+                                = static_cast<std::uintptr_t>((idx * this->m_dstPitchBytes).sum());
+                            assert(offsetBytes % alignof(Elem) == 0);
+                            Elem* elem = reinterpret_cast<Elem*>(
+                                __builtin_assume_aligned(this->m_dstMemNative + offsetBytes, alignof(Elem)));
+
+                            *elem = this->m_value;
+                        });
```

Comment:
Question for @psychocoderHPC: this approach should be correct for any padding, even between the outermost elements.
However, I expect that most buffers and views will have no padding for the outermost elements; for those we could write an optimised version that the compiler should be able to vectorise better.
As this is "just" an optimisation, is it OK if we add this in a future PR ?

Reply:
Yes optimizations can go in seperate PRs.
Assuming element alignment is IMO OK, I am wondering only about using `__builtin_assume_aligned()` because this is a gnu build in and I am not sure if all common compilers e.g. MSVC is supporting it.

IMO this function is here not doing anything because we ask for the alignment of the type but a pointer to the type is already type aligned.

Reply:
offline discussed: There there is a warning that we cast from lower alignment to higher with gcc  14 if we do not use the build-in.